egg/core/callbacks.py

```diff
@@ -131,22 +134,24 @@ def __init__(
         checkpoint_path: Union[str, pathlib.Path],
         checkpoint_freq: int = 1,
         prefix: str = "",
+        max_checkpoints: int = sys.maxsize,
     ):
+        """Saves a checkpoint file for training.
+        :param checkpoint_path:  path to checkpoint directory, will be created if not present
+        :param checkpoint_freq:  Number of epochs for checkpoint saving
+        :param prefix: Name of checkpoint file, will be {prefix}{current_epoch}.tar
+        :param max_checkpoints: Max number of concurrent checkpoint files in the directory.
+        """
         self.checkpoint_path = pathlib.Path(checkpoint_path)
         self.checkpoint_freq = checkpoint_freq
         self.prefix = prefix
+        self.max_checkpoints = max_checkpoints
         self.epoch_counter = 0
```

Comment:
Could you please remove this? I don't think we really need. The trainer instance has the epoch counter anyway

Reply:
Do you mean the epoch_counter?
it is used when getting the checkpoint [here](https://github.com/facebookresearch/EGG/blob/edde9f5628dde520bd58f728991091bb65d51f6b/egg/core/callbacks.py#L174)

Reply:
On the other hand if you are referring to max_checkpoints it is needed to keep track of the max number of checkpoints to save

Reply:
Sorry my bad, you're right!